packages/@cdktf/provider-generator/lib/get/generator/emitter/attributes-emitter.ts

```diff
@@ -21,7 +24,12 @@ export class AttributesEmitter {
     const hasInputMethod = isStored;
 
     if (isStored) {
-      this.code.line(`private ${att.storageName}?: ${att.type.storedName}; `);
+      this.code.line(
+        `private ${att.storageName}?: ${
+          // TODO: this is a hack to make optional attributes required on resources but not on structs
+          this.context === "resource" ? att.type.name : att.type.storedName
```

Comment:
What is the case where this check is needed? Since we are always are emitting as an optional field and the only difference between `type.name` and `type.storedName` is a possible `| undefined` it doesn't seem like it changes anything.

Reply:
I remove this, not sure why I had it in the first place. If something arises I'll add a test to catch it 👍